File: Examples/OneSignalUserDemo/userapp/src/main/java/com/onesignal/usersdktest/activity/MainActivity.java
package com.onesignal.usersdktest.activity;

import android.os.Build;
import android.os.Bundle;

import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;

import com.onesignal.onesignal.OneSignal;
import com.onesignal.usersdktest.R;
import com.onesignal.usersdktest.model.MainActivityViewModel;

@RequiresApi(api = Build.VERSION_CODES.N)
public class MainActivity extends AppCompatActivity {

    private MainActivityViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_activity_layout);

        viewModel = new MainActivityViewModel();
        OneSignal.getNotifications().addPushPermissionHandler(viewModel);
// TODO("STILL SUPPORT?")
//        OneSignal.addSubscriptionObserver(viewModel);
//        OneSignal.addEmailSubscriptionObserver(viewModel);
        viewModel.onActivityCreated(this)
                .setupInterfaceElements();
    }

    @Override
    public void onBackPressed() {
        if (!viewModel.scrollToTopIfAvailable()) {
            super.onBackPressed();
            overridePendingTransition(R.anim.fade_in, R.anim.fade_out);
        }
    }

    @Override
    protected void onResume() {
        super.onResume();

        boolean hasConsent = OneSignal.getRequiresPrivacyConsent();
        if (hasConsent)
            viewModel.setupSettingsLayout();
    }
}

File: examples/demo/app/src/main/java/com/onesignal/sdktest/ui/main/MainActivity.kt
package com.onesignal.sdktest.ui.main

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.viewModels
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.Surface
import androidx.compose.ui.Modifier
import com.onesignal.sdktest.ui.theme.LightBackground
import com.onesignal.sdktest.ui.theme.OneSignalTheme

class MainActivity : ComponentActivity() {

    private val viewModel: MainViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContent {
            OneSignalTheme {
                Surface(
                    modifier = Modifier.fillMaxSize(),
                    color = LightBackground
                ) {
                    MainScreen(viewModel = viewModel)
                }
            }
        }
    }
}
